### PR TITLE
some small improvements and a basic command line interface

### PR DESCRIPTION
**Pull request by LunarEclipse363**
the commit messages should be fairly self explanatory, but:

- the editorconfig can be added to gitignore instead if desired
- the cli is very simple and only works with stdin/stdout for now, to use it `cargo run --features=cli`
- there is a FIXME comment left in a part of the spec where something looks missing

### CODE REVIEW

File: spec.doll [hunk 1]
```diff
@@ -61,7 +61,7 @@
 				block content allows far more flexibility in the content of tags, being able to put anything inside of them,
 				without markdoll interpreting it whatsoever, leaving that to that tag
 
-				however, this requires that the content
+				however, this requires that the content [//:FIXME\: what is missing here?]
```
Comment (0x57e11a) [inline spec.doll]: :neocat_floof_explode:

it believes that it was supposed to say something along the lines "however, this requires that the content is indented one level further than the tag, to let the parser know what is and is not part of the content"

oops :3

File: src/main.rs
Comment (0x57e11a) [inline src/main.rs:1]: perfect! love that the stdout/stderr are separated to allow piping, and that on fail, it returns `1` <3

File: README.md [hunk 2]
```diff
@@ -12,3 +12,7 @@ to get an idea of the syntax, [visit the repository](https://github.com/0x57e11a
 
 - `ariadne`
   allows fancy diagnostic printing, requires `std`
+
+## minimum supported rust version
+
+this library requires features from rust 1.81
```
Comment (0x57e11a) [inline README.md:18]: excuse it for being ignorant, but what features require 1.81? not a demand to change it, just curious :3

File: Cargo.toml
Comment (0x57e11a) [inline Cargo.toml:1]: adding `rust-version = "1.81"` should help with consistency with the README

**Review by 0x57e11a — CHANGES_REQUESTED** [review]
mostly just admiration/curiosity, with a couple small changes :3

File: README.md (hunk 2 above)
Comment (LunarEclipse363) [reply README.md:18]: IIRC it was reasons for `#![allow...]`

When I first tried running cargo check with 1.79, it was also missing the `Default` impl for `Rc<str>`, not sure if that was added in 1.80 or 1.81 though.

Comment (0x57e11a) [reply README.md:18]: sounds good! just curious!

**Review by 0x57e11a — APPROVED** [review]
beep

**Pull request merged**